frontend/packages/data-portal/app/routes/api.deposition-datasets.ts

```diff
@@ -0,0 +1,88 @@
+import { LoaderFunctionArgs } from '@remix-run/server-runtime'
+
+import {
+  getDatasetsForDepositionViaAnnotationShapes,
+  getDatasetsForDepositionViaTomograms,
+} from 'app/graphql/getDatasetsForDepositionV2.server'
+import { DataContentsType } from 'app/types/deposition-queries'
+import { aggregateDatasetInfo } from 'app/utils/deposition-aggregation'
+
+interface DatasetOption {
+  id: number
+  title: string
+  organismName: string | null
+}
+
+interface DepositionDatasetsResponse {
+  datasets: DatasetOption[]
+  organismCounts: Record<string, number>
+  annotationCounts: Record<number, number>
+  tomogramCounts: Record<number, number>
+}
+
+export async function loader({ request }: LoaderFunctionArgs) {
+  const url = new URL(request.url)
+  const depositionId = url.searchParams.get('depositionId')
+  const type = url.searchParams.get('type')
+
+  if (!depositionId || Number.isNaN(Number(depositionId))) {
+    return new Response('Missing or invalid depositionId', { status: 400 })
+  }
+
+  if (
+    !type ||
+    !Object.values(DataContentsType).includes(type as DataContentsType)
+  ) {
+    return new Response(
+      'Missing or invalid type parameter. Must be "annotations" or "tomograms"',
```

Comment:
Nit: Would be nice if it wasn't a literal string with `"annotations" or "tomograms"` in it, but rather was interpolated from the values of `DataContentsType`. Dunno if that's a huge pain to do though. I'm just thinking that this error message could get stale if we eventually support a third option, but this error message doesn't get updated.